File: Android/Currencyconverter/app/src/main/java/com/example/currencyconverter/MainActivity.java
package com.example.currencyconverter;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
 public void buttonclick(View view)
 {
     EditText amount1=findViewById(R.id.amount);
     TextView converted=findViewById(R.id.textView2);

     float x = Float.parseFloat(amount1.getText().toString());
     double y=x*3.64;
     converted.setText(String.valueOf(y)+" QR");

 }
    public void clearb(View view) {
        EditText amount1=findViewById(R.id.amount);
        TextView converted=findViewById(R.id.textView2);

        amount1.setText(" ");
        converted.setText(" ");
    }
}

File: Android/Retrofit demo/app/src/main/java/com/example/retrofittut/MainActivity.kt
package com.example.retrofittut

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.core.view.isVisible
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.retrofittut.databinding.ActivityMainBinding
import retrofit2.HttpException
import java.io.IOException
const val TAG = "MainAcctivity"
class MainActivity : AppCompatActivity() {


    lateinit var binding: ActivityMainBinding
    private lateinit var todoAdapter : TodoAdapter
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        setupRecyclerView()

        lifecycleScope.launchWhenCreated {
            binding.progressBar.isVisible = true

            val response = try {
                RetrofitInstance.api.getTodos()
            }catch (e : IOException){
                binding.progressBar.isVisible = false
                Log.e(TAG,"internet expection")
                return@launchWhenCreated
            }catch (e : HttpException){
                binding.progressBar.isVisible = false
                Log.e(TAG,"http expection , unexpexted")
                return@launchWhenCreated
            }

           if (response.isSuccessful && response.body() != null ){
               todoAdapter.todos  = response.body()!!
           }else {
               Log.e(TAG, "Response not successful")
           }
            binding.progressBar.isVisible = false
        }

    }

    private fun setupRecyclerView() = binding.rvTodos.apply {
        todoAdapter = TodoAdapter()
        adapter = todoAdapter
        layoutManager = LinearLayoutManager(this@MainActivity)

    }
}